src/test/java/org/folio/tlib/postgres/PgCqlQueryTest.java

```diff
@@ -0,0 +1,125 @@
+    String[][] list = new String[][] {
+        { "(", "error: expected index or term, got EOF" },
+        { "foo=bar", "error: Unsupported CQL index: foo" },
+        { "Title=v1", ftResponse("title", "v1") },
+        { "Title all v1", ftResponse("title", "v1") },
+        { "Title>v1", "error: Unsupported operator > for: Title > v1" },
+        { "Title=\"men's room\"", ftResponse("title", "men''s room") },
+        { "Title=men's room", ftResponse("title", "men''s room") },
+        { "Title=v1*", "error: Masking op * unsupported for: Title = v1*" },
+        { "Title=v1?", "error: Masking op ? unsupported for: Title = v1?" },
+        { "Title=v1^", "error: Anchor op ^ unsupported for: Title = v1^" },
+        { "Title=a\\*b", ftResponse("title", "a*b") },
+        { "Title=a\\^b", ftResponse("title", "a^b") },
+        { "Title=a\\?b", ftResponse("title", "a?b") },
+        { "Title=a\\?b", ftResponse("title", "a?b") },
+        { "Title=a\\n", ftResponse("title", "a\\n") },
+        { "Title=\"a\\\"\"", ftResponse("title", "a\"") },
+        { "Title=\"a\\\"b\"", ftResponse("title", "a\"b") },
+        { "Title=a\\12", ftResponse("title", "a\\12") },
+        { "Title=a\\\\", ftResponse("title", "a\\") },
+        { "Title=a\\'", ftResponse("title", "a\\''") },
+        { "Title=a\\'b", ftResponse("title", "a\\''b") },
+        { "Title=a\\\\\\n", ftResponse("title", "a\\\\n") },
+        { "Title=a\\\\", ftResponse("title", "a\\") },
+        { "Title=aa\\\\1", ftResponse("title", "aa\\1") },
+        { "Title=ab\\\\\\?", ftResponse("title", "ab\\?") },
+        { "Title=\"b\\\\\"", ftResponse("title", "b\\") },
+        { "Title=\"c\\\\'\"", ftResponse("title", "c\\''") },
+        { "Title=\"c\\\\d\"", ftResponse("title", "c\\d") },
+        { "Title=\"d\\\\\\\\\"", ftResponse("title", "d\\\\") },
+        { "Title=\"x\\\\\\\"\\\\\"", ftResponse("title", "x\\\"\\") },
+        { "Title=\"\"", "title IS NULL" },
+        { "Title<>\"\"", "title IS NOT NULL" },
+        { "Title==\"\"", "title = ''" },
+        { "Title==\"*?^\"", "title = '*?^'" },
+        { "Title==\"\\*\\?\\^\"", "title = '\\*\\?\\^'" },
+        { "Title==\"b\\\\\"", "title = 'b\\'" },
+        { "Title==\"c\\\\'\"", "title = 'c\\'''" },
+        { "Title==\"d\\\\\\\\\"", "title = 'd\\\\'" },
+        { "Title==\"e\\\\\\\"\\\\\"", "title = 'e\\\"\\'" },
+        { "Title>\"\"", "error: Unsupported operator > for: Title > \"\"" },
+        { "Title==v1 or title==v2",  "(title = 'v1' OR title = 'v2')"},
+        { "isbn=978-3-16-148410-0", "isbn = '978-3-16-148410-0'" },
+        { "isbn=978-3-16-148410-*", "isbn = '978-3-16-148410-*'" },
+        { "cql.allRecords=1 or title==v1", null },
+        { "title==v1 or cql.allRecords=1", null },
+        { "Title==v1 and title==v2", "(title = 'v1' AND title = 'v2')" },
+        { "Title==v1 and cql.allRecords=1", "title = 'v1'" },
+        { "cql.allRecords=1 and Title==v2", "title = 'v2'" },
```

Comment:
CQL and SQL have different operator precedence:

CQL:
> Boolean Operators
> and, or, not and prox.
> Boolean operators all have the same precedence; they are evaluated left-to-right.

https://www.loc.gov/standards/sru/cql/spec.html

SQL:
> Operator Precedence (highest to lowest)
> NOT
> AND
> OR

https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-PRECEDENCE-TABLE

Please add unit tests that show that there are enough braces to stop SQL operator precedence from kicking in.

Reply:
Added some now.